File: src/main/java/java_spark/spark_book/第3章/action_operate_RDD.java
package java_spark.spark_book.第3章;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

import java.util.Iterator;
import java.util.List;

public class action_operate_RDD {
    public static void main(String[] args){
        SparkConf conf=new SparkConf().setAppName("action_operate_RDD").setMaster("local");
        JavaSparkContext sc=new JavaSparkContext(conf);

        JavaRDD<String> inputRDD=sc.textFile("/home/hdc/data/word1");
        JavaRDD<String> hdcRDD=inputRDD.filter((Function<String, Boolean>)s->{
            System.out.println("S:"+s);
            return s.contains("hdc");
        });
        JavaRDD<String> amRDD=inputRDD.filter(new Function<String, Boolean>() {
            public Boolean call(String s) throws Exception {
                return s.contains("am");
            }
        });

        JavaRDD<String> hdc_am_RDD=hdcRDD.union(amRDD);
        List<String>list=hdc_am_RDD.collect();
        Iterator<String> iterator=list.iterator();
        System.out.println("------------------------------------------------");
        while (iterator.hasNext())
            System.out.println(iterator.next());
        System.out.println("------------------------------------------------");
        System.out.println("hdc_am_RDD："+hdc_am_RDD.collect().toString());
    }
}

File: src/main/scala/scala_spark/spark_book/第3章/action_operate_RDD.scala
package scala_spark.spark_book.第3章

import org.apache.spark.{SparkConf, SparkContext}

object action_operate_RDD {
  def main(args:Array[String]):Unit={
    val conf=new SparkConf().setAppName("action_operate_RDD").setMaster("local")
    val sc=new SparkContext(conf)

    val inputRDD=sc.textFile("/home/hdc/word")
    val hdcRDD=inputRDD.filter(line=>line.contains("hdc"))
    val amRDD=inputRDD.filter(line=>line.contains("am"))
    val hdc_am_RDD=hdcRDD.union(amRDD)
    //RDD.collect().mkString("")输出RDD的内容
    //collect不能用在大规模数据集上。
    println("hdcRDD：\n"+hdcRDD.collect().mkString("\n"))
    println("amRDD:\n"+amRDD.collect().mkString("\n"))
    println("hdc_am_RDD：\n"+hdc_am_RDD.collect().mkString("\n"))
    //取出一定数量的RDD中元素
    println(hdc_am_RDD.take(2).foreach(println(_)))

  }

}
